packages/compass-data-modeling/src/services/data-model-storage-atlas.tsx

```diff
@@ -0,0 +1,119 @@
+import { AtlasUserData } from '@mongodb-js/compass-user-data';
+import type {
+  DataModelStorage,
+  MongoDBDataModelDescription,
+} from './data-model-storage';
+import { MongoDBDataModelDescriptionSchema } from './data-model-storage';
+import dataModelStorageInMemory from './data-model-storage-in-memory';
+import {
+  atlasServiceLocator,
+  type AtlasService,
+} from '@mongodb-js/atlas-service/provider';
+import { createServiceProvider } from '@mongodb-js/compass-app-registry';
+import { DataModelStorageServiceProvider } from '../provider';
+import React, { useRef } from 'react';
+import { mongoLogId, useLogger } from '@mongodb-js/compass-logging/provider';
+
+class DataModelStorageAtlas implements DataModelStorage {
+  private readonly userData: AtlasUserData<
+    typeof MongoDBDataModelDescriptionSchema
+  >;
+  constructor(orgId: string, projectId: string, atlasService: AtlasService) {
+    this.userData = new AtlasUserData(
+      MongoDBDataModelDescriptionSchema,
+      'dataModelDescriptions',
+      {
+        orgId,
+        projectId,
+        getResourceUrl(path) {
+          // TODO(COMPASS-9960): this is copied from compass-web entrypoint for
+          // brevity, but shouldn't be defined outside of AtlasUserData, this
+          // logic is literally the same between all user data instances and can
+          // be encapsulated inside of the AtlasUserData class implementation
```

Comment:
This clean-up will touch a lot of unrelated code, so I'm splitting it to a separate PR